{"name": "BTC普通账户转账", "status": "broken", "statusDetails": {"message": "UnboundLocalError: local variable 'url' referenced before assignment", "trace": "self = <test_success.Test_transfer_success object at 0x10e4e6940>, test_title = 'BTC普通账户转账'\nprivatekey = ['b61190f8b2fcb4754ff8527f083bb3df68bf89285d7c5e428276f5f4572b7abe'], networkCode = 'BTC'\nfromaddress = 'tb1qhcuul2fcyv5rvr5mc5cy5saz0q5p3kx0gsy3c0', toaddress = 'tb1qhcuul2fcyv5rvr5mc5cy5saz0q5p3kx0gsy3c0', value = '100'\n\n    @allure.story(\"Transfer Success!\")\n    @allure.title('{test_title}')\n    @pytest.mark.parametrize('test_title,privatekey,networkCode,fromaddress,toaddress,value', test_data)\n    def test_custodial(self,test_title,privatekey,networkCode,fromaddress,toaddress,value):\n    \n        with allure.step(\"查询地址UTXO\"):\n>           utxo = Httpcore.HttpCoreUtils.query_utxo(\"networkCode\",\"fromaddress\")\n\nTestcasecore/Test_Transactions/Test_Transfer/Test_Transfer_BTC/test_success.py:34: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nnetworkCode = 'networkCode', address = 'fromaddress'\n\n    @staticmethod\n    # 查询UTXO\n    def query_utxo(networkCode:str, address:str):\n        if networkCode == \"BTC\":\n            url = \"https://api.bitcore.io/api/BTC/testnet/address/\" + address + \"?unspent=true\"\n        elif networkCode == \"DOGE\":\n            url = \"https://api.bitcore.io/api/DOGE/mainnet/address/\" + address + \"?unspent=true\"\n        else:\n            logger.error(\"networkCode not support!\")\n        headers = {\n            \"Content-Type\": \"application/json\"\n        }\n    \n>       logger.info('\\n'+\"<-----Query UTXO----->\"+\"\\n\"+\"Url:\"+url+'\\n\\n'+'Headers:'+json.dumps(headers)+'\\n\\n')\nE       UnboundLocalError: local variable 'url' referenced before assignment\n\nCommon/Httpcore.py:224: UnboundLocalError"}, "steps": [{"name": "查询地址UTXO", "status": "broken", "statusDetails": {"message": "UnboundLocalError: local variable 'url' referenced before assignment\n", "trace": "  File \"/Users/lilong/Documents/Test_Api/Testcasecore/Test_Transactions/Test_Transfer/Test_Transfer_BTC/test_success.py\", line 34, in test_custodial\n    utxo = Httpcore.HttpCoreUtils.query_utxo(\"networkCode\",\"fromaddress\")\n  File \"/Users/lilong/Documents/Test_Api/Common/Httpcore.py\", line 224, in query_utxo\n    logger.info('\\n'+\"<-----Query UTXO----->\"+\"\\n\"+\"Url:\"+url+'\\n\\n'+'Headers:'+json.dumps(headers)+'\\n\\n')\n"}, "start": 1669947510312, "stop": 1669947510313}], "attachments": [{"name": "log", "source": "3fbdc739-2e67-4c0b-a561-7a99687b2a7f-attachment.txt", "type": "text/plain"}], "parameters": [{"name": "test_title", "value": "'BTC普通账户转账'"}, {"name": "privatekey", "value": "['b61190f8b2fcb4754ff8527f083bb3df68bf89285d7c5e428276f5f4572b7abe']"}, {"name": "networkCode", "value": "'BTC'"}, {"name": "fromaddress", "value": "'tb1qhcuul2fcyv5rvr5mc5cy5saz0q5p3kx0gsy3c0'"}, {"name": "toaddress", "value": "'tb1qhcuul2fcyv5rvr5mc5cy5saz0q5p3kx0gsy3c0'"}, {"name": "value", "value": "'100'"}], "start": 1669947510312, "stop": 1669947510314, "uuid": "f055ee48-ae0e-4293-b475-eac633749eb8", "historyId": "df7a9ec3a68d9348ec991e3f399c784f", "testCaseId": "492d3f1d90543904ec0223cce54ea25b", "fullName": "Testcasecore.Test_Transactions.Test_Transfer.Test_Transfer_BTC.test_success.Test_transfer_success#test_custodial", "labels": [{"name": "feature", "value": "Transfer Success!"}, {"name": "story", "value": "Transfer Success!"}, {"name": "parentSuite", "value": "Testcasecore.Test_Transactions.Test_Transfer.Test_Transfer_BTC"}, {"name": "suite", "value": "test_success"}, {"name": "subSuite", "value": "Test_transfer_success"}, {"name": "host", "value": "lilongdeMacBook-Pro.local"}, {"name": "thread", "value": "51144-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "Testcasecore.Test_Transactions.Test_Transfer.Test_Transfer_BTC.test_success"}]}